src/main/java/org/truffleruby/core/proc/ProcNodes.java

```diff
@@ -205,7 +205,7 @@ public int arity(DynamicObject proc) {
         @Specialization
         public DynamicObject binding(DynamicObject proc) {
             final MaterializedFrame frame = Layouts.PROC.getDeclarationFrame(proc);
-            return BindingNodes.createBinding(getContext(), frame);
+            return frame == null ? nil() : BindingNodes.createBinding(getContext(), frame);
```

Comment:
This might have bad performance effects, like preventing EA of the Binding.
Is it needed?

Reply:
Blocks created from methods don't have a declaration frame. I could change it to two specializations and a guard if that would be better.

Reply:
Do you have such a case a Method#to_proc in this PR?
It might be easier/cleaner to just give a declaration frame for Method#to_proc for consistency.
Two specializations would definitely be better.

Reply:
Just checked using `irb` and in MRI procs always have a binding, so I'd better fix that so we match.

Reply:
So then the solution is to create a declaration frame in Method#to_proc too then?
If this is not needed in this PR it would be nicer to extract in its own PR.